zetaclient/tss/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,68 @@
+package ratelimit
+
+import (
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/semaphore"
+)
+
+// RateLimiter is a simple semaphore for limiting the number of concurrent signatures.
+//
+// This is a naive implementation that probably requires
+// improvements to handle real-world scenarios.
+//
+// Pros:
+// - Has a simple interface that hides the underlying implementation details;
+//
+// Cons:
+// - Doesn't take number of signatures per chain into account;
+// - Doesn't take nonce ordering into account;
+// - Doesn't take chain "fairness" into account;
+//
+// TBD:
```

Comment:
We could get the value form an on-chain param instead of the config.

Coordinating to update across all network would be easier